backend/commons/src/main/java/org/sonarsource/sonarlint/core/commons/progress/TaskManager.java

```diff
@@ -28,10 +28,20 @@ public class TaskManager {
   private static final ProgressMonitor NO_OP = new NoOpProgressMonitor();
   private final ConcurrentHashMap<String, ProgressMonitor> progressMonitorsByTaskId = new ConcurrentHashMap<>();
 
-  public final void runTask(@Nullable String configurationScopeId, UUID taskId, String title, @Nullable String message, boolean indeterminate, boolean cancellable, Task task,
+  public final void createAndRunTask(@Nullable String configurationScopeId, UUID taskId, String title, @Nullable String message,
+    boolean indeterminate, boolean cancellable, Task task,
     SonarLintCancelMonitor cancelMonitor) {
-    var progressMonitor = startProgress(configurationScopeId, taskId, title, message, indeterminate, cancellable, cancelMonitor);
-    progressMonitorsByTaskId.put(taskId.toString(), progressMonitor);
+    createTask(taskId, cancelMonitor);
+    runExistingTask(configurationScopeId, taskId, title, message, indeterminate, cancellable, task, cancelMonitor);
+  }
+
+  public final void runExistingTask(@Nullable String configurationScopeId, UUID taskId, String title, @Nullable String message,
+    boolean indeterminate, boolean cancellable, Task task, SonarLintCancelMonitor cancelMonitor) {
+    var progressMonitor = progressMonitorsByTaskId.get(taskId.toString());
+    if (progressMonitor == null) {
+      return;
+    }
```

Comment:
We should maybe display at least a log here, it's not normal to not find the progress monitor in the map